frontend/packages/sdk/src/OnChainRegistry.ts

```diff
@@ -269,22 +290,78 @@ export class OnChainRegistry {
           throw new Error('No cluster found.')
         }
         const [clusterId, clusterInfo] = clusters[0]
-        const workers = await this.getClusterWorkers(clusterId)
-        this.#phactory = await this.preparePruntimeClientOrThrows(workers[0].endpoints.default)
+        const [workerId, endpoint, phactory] = await ackFirst()(this.api, clusterId)
+        this.#phactory = phactory
         this.clusterId = clusterId
         this.clusterInfo = clusterInfo
-        this.workerInfo = workers[0]
+        this.workerInfo = {
+          pubkey: workerId,
+          clusterId: clusterId,
+          endpoints: {
+            default: endpoint,
+            v1: [endpoint],
+          },
+        }
         this.#ready = true
         await this.prepareSystemOrThrows(clusterInfo)
         return
       }
 
       // Scenario 2: connect to specified worker.
       if (args.length === 1 && args[0] instanceof Object) {
+        if (args[0].strategy) {
+          let clusterId = args[0].clusterId
+          let clusterInfo
+          if (!clusterId) {
+            const clusters = await this.getAllClusters()
+            if (!clusters || clusters.length === 0) {
+              throw new Error('No cluster found.')
+            }
+            ;[clusterId, clusterInfo] = clusters[0]
+          } else {
+            clusterInfo = await this.getClusterInfoById(clusterId)
+            if (!clusterInfo) {
+              throw new Error(`Cluster not found: ${clusterId}`)
+            }
+          }
+          if (args[0].strategy === 'ack-first') {
+            const [workerId, endpoint, phactory] = await ackFirst()(this.api, clusterId)
+            this.#phactory = phactory
+            this.clusterId = clusterId
+            this.clusterInfo = clusterInfo
+            this.workerInfo = {
+              pubkey: workerId,
+              clusterId: clusterId,
+              endpoints: {
+                default: endpoint,
+                v1: [endpoint],
+              },
+            }
+            this.#ready = true
+            await this.prepareSystemOrThrows(clusterInfo)
```

Comment:
A checked worker may still throw an error here, should move it into checker.

Reply:
I thinking it should how we do healthy check, so it's ok here.